temporalio/lib/temporalio/internal/worker/workflow_instance.rb

```diff
@@ -308,11 +317,26 @@ def create_instance
           @inbound.init(OutboundImplementation.new(self))
 
           # Create the user instance
-          if @definition.init
-            @definition.workflow_class.new(*@workflow_arguments)
-          else
-            @definition.workflow_class.new
+          instance = if @definition.init
+                       @definition.workflow_class.new(*@workflow_arguments)
+                     else
+                       @definition.workflow_class.new
+                     end
+
+          # Run Dynamic config getter
+          unless @definition.dynamic_options_method.nil?
+            dynamic_options = instance.send @definition.dynamic_options_method
           end
+          unless dynamic_options.nil?
+            unless dynamic_options.versioning_behavior.nil?
+              @definition_options.versioning_behavior = dynamic_options.versioning_behavior
+            end
+            unless dynamic_options.failure_exception_types.nil?
+              @definition_options.failure_exception_types = dynamic_options.failure_exception_types
+            end
+          end
```

Comment:
A few things I think could be clearer here. Would rewrite as:

```ruby
if @definition.dynamic_options_method
  dynamic_options = instance.send(@definition.dynamic_options_method)
  @definition_options.versioning_behavior = dynamic_options.versioning_behavior if dynamic_options&.versioning_behavior
  @definition_options.failure_exception_types = dynamic_options.failure_exception_types if dynamic_options&.failure_exception_types
end
```

Reply:
Yeah I absolutely hated writing this, and rubocop kep suggesting dumb shit like early- returns that only made sense if you weren't ever going to add anything after it. Thanks for the suggestion.

Reply:
Rubocop multilined this anyway too, but, still butter than before.